app/static/tests/e2e/code.etest.ts

```diff
@@ -47,36 +47,40 @@ test.beforeEach(async ({ page }) => {
-        const glyphElement = await page.locator(`.margin-view-overlays div:nth-child(${i + 1}) >> div`);
         if (i === line - 1) {
-            expect(lineElement).toHaveClass(`cdr editor-${state}-background`);
-            expect(glyphElement.nth(0)).toHaveClass(`cgmr codicon ${editorGlyphs[state]} ${state}-glyph-style ms-1`);
-            expect(glyphElement.nth(1)).toHaveClass("line-numbers lh-odd");
+            await expect(lineElement).toHaveClass(`cdr editor-${state}-background`);
         } else if (i === numOfLines - 1) {
-            expect(lineElement).toHaveClass("current-line");
-            expect(glyphElement.nth(0)).toHaveClass("current-line current-line-margin-both");
-            expect(glyphElement.nth(1)).toHaveClass(/\bactive-line-number\b/);
-            expect(glyphElement.nth(1)).toHaveClass(/\bline-numbers\b/);
-            expect(glyphElement.nth(1)).toHaveClass(/\blh-odd\b/);
+            await expect(lineElement).toHaveClass("current-line");
         } else {
-            expect(lineElement).toHaveCount(0);
-            expect(glyphElement).toHaveClass("line-numbers lh-odd");
+            await expect(lineElement).toHaveCount(0);
         }
     }
+
+    const lineHeight = await page.locator(".margin-view-overlays").evaluate(el => {
+        return window.getComputedStyle(el).getPropertyValue("line-height");
+    });
+    const glyphTop = await page.locator(`.${editorGlyphs[state]}`).evaluate(el => {
+        return window.getComputedStyle(el).getPropertyValue("top");
+    });
+    expect(glyphTop).toBe(`${(line - 1) * parseInt(lineHeight)}px`);
 };
 
-const expectMonacoHover = async (type: "glyph" | "content", line: number, message: string, page: any) => {
+const expectMonacoHover = async (type: "glyph" | "content", line: number, message: string, page: Page) => {
     const isGlyph = type === "glyph";
-    const hoverElem = `.${isGlyph ? "margin-" : ""}view-overlays div:nth-child(${line}) >> div`;
-    const tooltipElem = `${isGlyph ? ".overlayWidgets" : ".overflowingContentWidgets"} .hover-contents div p`;
+    const hoverElem = type === "content" ?
+        `.view-overlays div:nth-child(${line}) >> div` :
+        ".glyph-margin-widgets .fa-solid";
+    // const hoverElem = `.${isGlyph ? "margin-" : ""}view-overlays div:nth-child(${line}) >> div`;
+    // const tooltipElem = `${isGlyph ? ".overlayWidgets" : ".overflowingContentWidgets"} .hover-contents div p`;
     await page.hover(hoverElem, { force: true });
-    const tooltip = await page.locator(tooltipElem);
-    await tooltip.waitFor({ timeout: 1000 });
-    await expect(tooltip).toHaveText(message);
-    await expect(tooltip).toHaveCSS("visibility", "visible");
+    // const tooltip = await page.locator(tooltipElem);
+    // await tooltip.waitFor({ timeout: 2000 });
+    // await expect(tooltip).toHaveText(message);
+    // await expect(tooltip).toHaveCSS("visibility", "visible");
```

Comment:
ignore the commented out code and the block below, i remove them in the next PR, i accidentally pushed the cleaning out commented code commit to the next pr